src/OrchardCore/OrchardCore.Abstractions/Json/Serialization/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,13 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace System.Text.Json.Serialization;
+
+public static class ServiceCollectionExtensions
+{
+    /// <summary>
+    /// Registers a JSON type resolver allowing to serialize a given type from its base type.
+    /// </summary>
+    public static IServiceCollection AddJsonDerivedTypeInfo<TDerived, TBase>(this IServiceCollection services)
+        where TDerived : class where TBase : class =>
+        services.AddSingleton<IJsonDerivedTypeInfo, JsonDerivedTypeInfo<TDerived, TBase>>();
```

Comment:
This should be transient. Cheaper than singleton (won't be stored in root container), and is resolved once during the app (on startup).